indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -2438,9 +2438,19 @@ public int getNoticesQueueSize()
           builder.put(partition, makeSequenceNumber(sequence, useExclusiveStartSequenceNumberForNonFirstSequence()));
         }
       } else {
-        // if we don't have a startingOffset (first run or we had some previous failures and reset the sequences) then
-        // get the sequence from metadata storage (if available) or Kafka/Kinesis (otherwise)
-        OrderedSequenceNumber<SequenceOffsetType> offsetFromStorage = getOffsetFromStorageForPartition(partition);
+        OrderedSequenceNumber<SequenceOffsetType> offsetFromStorage;
+        try {
+          // if we don't have a startingOffset (first run or we had some previous failures and reset
+          // the sequences) then get the sequence from metadata storage (if available) or
+          // Kafka/Kinesis (otherwise)
+          offsetFromStorage = getOffsetFromStorageForPartition(partition);
+        }
+        catch (ISE iseException) {
+          log.warn(
+              "Could not fetch offset for partition [%s], sequence [%s] - skipping this partition",
+              partition, sequence);
+          continue;
```

Comment:
This means that if there is any error in getting the offset for a partition, we'll act like it doesn't exist. It works well in empty-partition case but would be confusing in any other case (or even wrong -- a transient error would cause us to skip a partition). Is there any way we can differentiate an empty partition from some sort of other error?